src/app/components/login/login.component.ts

```diff
@@ -0,0 +1,127 @@
+import {ChangeDetectionStrategy, Component, OnDestroy, OnInit, ViewEncapsulation} from '@angular/core';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
+import {AuthFacade} from '../../states/facade/auth.facade';
+import {buttonClickedAnimation} from '../../animations/button-click.animation';
+import {IInputControl} from '../../models/interfaces/input-control.model.i';
+import {IUser} from '../../models/interfaces/user.model.i';
+import {Subject} from 'rxjs';
+import {take, takeUntil} from 'rxjs/operators';
+import {Language} from '../../models/configs/language-options.model';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+  animations: [
+    buttonClickedAnimation
+  ],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.Default,
+  host: {
+    class: 'app-login'
+  }
+})
+
+export class LoginComponent implements OnInit, OnDestroy {
+  private readonly _onDestroy = new Subject();
+  loginAndRegisterForm: FormGroup;
+  errors: string;
+  email: IInputControl;
+  password: IInputControl;
+  registerClicked = false;
+  loginClicked = false;
```

Comment:
bitte auch hier die types angeben. TSLint meckert dann zwar, aber das kannst du in der tslint.json ausstellen

```suggestion
  registerClicked: boolean = false;
  loginClicked: boolean = false;
```